### ISSUE TITLE
Linter should treat `void` functions as non-returning

### ISSUE THREAD

**Issue by lantua** [biomejs/biome#7237]
### Environment information

```bash
CLI:
  Version:                      2.2.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.5.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.20

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/useIterableCallbackReturn

### Playground link

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAZABvAFMAbwBtAGUAdABoAGkAbgBnACgAKQA6ACAAdgBvAGkAZAAgAHsAfQAKAGMAbwBuAHMAdAAgAGEAIAA9ACAAWwAxACwAIAAyACwAIAAzAF0AOwAKAGEALgBmAG8AcgBFAGEAYwBoACgAKAApACAAPQA%2BACAAdgBvAGkAZAAgADQAKQA7AAoAYQAuAGYAbwByAEUAYQBjAGgAKAAoACkAIAA9AD4AIABkAG8AUwBvAG0AZQB0AGgAaQBuAGcAKAApACkAOwAKAGEALgBtAGEAcAAoACgAKQAgAD0APgAgAHYAbwBpAGQAIAA0ACkACgBhAC4AbQBhAHAAKAAoACkAIAA9AD4AIABkAG8AUwBvAG0AZQB0AGgAaQBuAGcAKAApACkAOwA%3D

### Expected result

This playground code used to pass the linter in version 2.1.4. Both `forEach` calls are now failing.

Either `void` function calls should be treated as non-returning, or the explicit `void` operator should be an escape hatch. The linter should also flag the `map` arguments that return `void`. With 2.2.0 checks, it forces `forEach` to always use block statements `() => {}`, which are always forced into multi-line by the formatter.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
